app/models/user.rb

```diff
@@ -220,12 +220,12 @@ def self.upload_user_list(user_class, user_list, encoding)
       end
       imported = user_class.upsert_all(user_hash, unique_by: :user_name, returning: %w[id user_name]) \
 unless user_hash.empty?
-      unless imported.nil?
-        successful_imports = imported.rows.map { |x| [x[1]] }.flatten
-        imported_ids = imported.rows.map { |x| [x[0]] }.flatten
-      else
+      if imported.nil?
```

Comment:
Hey David, I know you didn't want me to add the nil check but I know exactly why this happens and it is because the user_hash is empty, thus the .upsert_all returns nil. As for the `successful_imports = []` and `imported_ids = []`, I added those because I used byebug with the master branch and found out that those are what the values would be with .import.

Reply:
You're right that the above line will never execute if `user_hash` is empty. However, this is a place where the code can be simplified. Notice that `user_hash` is computed from `users`, so we can actually return much earlier. Look for where `users` is computed, and after that point you can do a `return parsed if users.empty?`, which is actually an improvement since it saves all of the other work.

Looking at the code, `parsed` has its `:invalid_records` key set on line 214, so you'll want to move that earlier so that even with the early return, `parsed` has this key.

Reply:
(And then you can remove the `unless user_hash.empty?` guard from the previous line.)